src/azure-cli/azure/cli/command_modules/util/_help.py

```diff
@@ -21,16 +21,16 @@
 examples:
   - name: Get Audit log through Microsoft Graph
     text: >
-        az rest --method get --uri https://graph.microsoft.com/beta/auditLogs/directoryAudits
+        az rest --method get --url https://graph.microsoft.com/beta/auditLogs/directoryAudits
   - name: Update a Azure Active Directory Graph User's display name
     text: >
-        az rest --method patch --uri "https://graph.microsoft.com/v1.0/users/johndoe@azuresdkteam.onmicrosoft.com" --body "{\\"displayName\\": \\"jondoe2\\"}"
+        az rest --method patch --url "https://graph.microsoft.com/v1.0/users/johndoe@azuresdkteam.onmicrosoft.com" --body "{\\"displayName\\": \\"jondoe2\\"}"
   - name: Get a virtual machine
     text: >
-        az rest --method get --uri /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}?api-version=2019-03-01
```

Comment:
>/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}?api-version=2019-03-01 [](start = 35, length = 142)

[https://danielmiessler.com/study/difference-between-uri-url/](https://danielmiessler.com/study/difference-between-uri-url/)
Do we need to change the sample? URL is a subset of uri, resource id like "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}?api-version=2019-03-01" is more like an uri

Reply:
I previously made this change in PR https://github.com/Azure/azure-cli/pull/13067 and provided more explanations.

Reply:
`/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}?api-version=2019-03-01` is not a valid URI because it lacks the necessary protocol like `tel:`, `mailto:`. `?api-version=2019-03-01` is also not allowed in a URI. 

Internally we prefix the resource ID with ARM endpoint so that it finally becomes a URL.

Reply:
internally everything will become a URL, for this parameter, I don't think we should change the sample, "All URLs are URIs, but not all URIs are URLs"

---
In reply to: [436453724](https://github.com/Azure/azure-cli/pull/13849#discussion_r436453724) [](ancestors = 436453724)

Reply:
Changed as requested. Use `--uri` for resource ID and `--url` for full/complete URLs.